src/app/backend/deploy.go

```diff
@@ -224,3 +237,28 @@ func getLabelsMap(labels []Label) map[string]string {
 
 	return result
 }
+
+// Deploys an app based on the given yaml or json file.
+func DeployAppFromFile(spec *AppDeploymentFromFileSpec, client *client.Client) error {
+	// Creates temp file to use it in kubectl command
+	file, err := ioutil.TempFile("", spec.Name)
+	if err != nil {
+		return err
```

Comment:
Right now all those error messages are sent to the user with error popup. We might improve it by hiding some error messages coming form server.